### PR TITLE
Fix for concat v2 unit test failure in [INTEL MKL] enabled Tensorflow

### PR DESCRIPTION
**Pull request by jojivk73**
Fix for unit test failure in [INTEL MKL] Tensorflow
    //tensorflow/python/ops/parallel_for:array_test (concat_v2) 
The unit test has axis type : INT64 i not supported by MKLDNN.
The fix disallows the mapping to MKLDNN ops when any type other than INT32 is used for axis.

### CODE REVIEW

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 1]
```diff
@@ -1496,6 +1496,15 @@ class MklLayoutRewritePass : public GraphOptimizationPass {
     }
     return false;
   }
+  // For NKL-DNN only int32 supported for axis data type
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:1499]: Nit:
```suggestion
  // For oneDNN, only int32 is supported for axis data type
```

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 2]
```diff
@@ -1496,6 +1496,15 @@ class MklLayoutRewritePass : public GraphOptimizationPass {
...
+  static bool ConcatV2Rewrite(const Node *n) {
+    DataType T;
+    GetNodeAttr(n->def(),"Tidx", &T);
+    if (T != DT_INT32) {
+      return false;
+    }
+    return true;
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:1506]: Nit: We can simplify this to:
```suggestion
    return (T == DT_INT32);
```

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc [hunk 3]
```diff
@@ -4258,6 +4258,29 @@ TEST_F(MklLayoutPassTest, NodeRewrite_ConcatV2_DeviceTest) {
             "A->D:2;B->D;B:1->D:1;C->E;D->E:1");
 }
 
+
+TEST_F(MklLayoutPassTest, NodeRewrite_ConcatV2_idxtest) {
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass_test.cc:4262]: Nit:
```suggestion
TEST_F(MklLayoutPassTest, NodeRewrite_ConcatV2_IndexTest) {
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the fix and sorry for the delay! I have a few nits.

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc (hunk 3 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc:4262]: @penpornk  Hi, thank you for the code review. I am covering the developer and will do code change today.  --GZ

**Comment by gzmkl** [comment]
@penpornk  Update has been done per code review suggestions.

**Comment by gzmkl** [comment]
> All (the pull request submitter and all commit authors) CLAs are signed, **but** one or more commits were authored or co-authored by someone other than the pull request submitter.
> 
> We need to confirm that all authors are ok with their commits being contributed to this project. Please have them confirm that by leaving a comment that contains only `@googlebot I consent.` in this pull request.
> 
> _Note to project maintainer:_ There may be cases where the author cannot leave a comment, or the comment is not properly detected as consent. In those cases, you can manually confirm consent of the commit author(s), and set the `cla` label to `yes` (if enabled on your project).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Ftensorflow%2Ftensorflow%2Fpull%2F43573) for more info**.

@googlebot I consent 

Hi, the author ( @jojivk73, Jojimon) is my teammate at Intel. He is on vacation and thus cannot confirm before he is back to work next Monday (October 12 2020). So please do special process for this case as we are approaching TF 2.4 cutting date. 
Thanks!

**Review by penpornk — APPROVED** [review]
Thank you, @gzmkl, for stepping in! I'll address the CLA in a separate comment.

**Comment by penpornk** [comment]
Manually setting CLA to yes per @gzmkl's [comment](https://github.com/tensorflow/tensorflow/pull/43573#issuecomment-703853824).

**Pull request merged**